bin/omarchy-setup-cloudflare-dns

```diff
@@ -0,0 +1,195 @@
+#!/bin/bash
+
+# omarchy-setup-cloudflare-dns
+# Script to configure DNS settings for Omarchy installations
+# Provides options for Cloudflare, local/DHCP, or custom DNS setups
+
+set -e
+
+RESOLVED_CONF="/etc/systemd/resolved.conf"
+BACKUP_CONF="/etc/systemd/resolved.conf.omarchy-backup"
+
+show_help() {
+    cat << EOF
+omarchy-setup-cloudflare-dns - Configure DNS settings for Omarchy
+
+USAGE:
+    omarchy-setup-cloudflare-dns [OPTION]
+
+OPTIONS:
+    cloudflare      Set Cloudflare (1.1.1.1) as primary DNS with Google (8.8.8.8) as fallback
+    dhcp           Use DHCP-provided DNS (preserves local router DNS for development)
+    custom         Interactive setup for custom DNS servers
+    restore        Restore original DNS configuration from backup
+    status         Show current DNS configuration
+    help           Show this help message
+
+EXAMPLES:
+    omarchy-setup-cloudflare-dns cloudflare    # Use Cloudflare DNS
+    omarchy-setup-cloudflare-dns dhcp          # Use router/DHCP DNS (recommended for local dev)
+    omarchy-setup-cloudflare-dns custom        # Set custom DNS servers
+    omarchy-setup-cloudflare-dns status        # Check current configuration
+
+NOTE:
+    - This script requires sudo privileges to modify system DNS settings
+    - Changes take effect after restarting systemd-resolved
+    - A backup of your current configuration is automatically created
+EOF
+}
+
+backup_current_config() {
+    if [[ -f "$RESOLVED_CONF" ]] && [[ ! -f "$BACKUP_CONF" ]]; then
+        echo "Creating backup of current DNS configuration..."
+        sudo cp "$RESOLVED_CONF" "$BACKUP_CONF"
+        echo "Backup saved to: $BACKUP_CONF"
+    fi
+}
+
+restart_resolved() {
+    echo "Restarting systemd-resolved to apply changes..."
+    sudo systemctl restart systemd-resolved
+    echo "DNS configuration updated successfully!"
+}
+
+setup_cloudflare() {
```

Comment:
One thing I realized with the original version we had was that the DNS changes didn't actually apply to interfaces that set their own DNS to come of DHCP. Now that we're making this entirely optional, setting cloudflare dns should force that DNS onto all of the normal ethernet and wifi interfaces (but not the docker / tailnet interfaces).